Add support for pre-compiling angular templates when doing a dist build.

agxs opened the pull request:
Adds a new task to use the Angular NGC compiler when running serve-dist or compile tasks. This pre-compiles the angular templates so they load much faster.

Also makes sure to exclude the ngfactories from tslint.

agxs committed:
Add support for pre-compiling angular templates when doing a dist build.

marksmall commented on src/gulp/tasks/ts-lint.js:
```diff
@@ -17,6 +18,7 @@ class TsLintTaskLoader extends AbstractTaskLoader {
 
         gulp.task("ts-lint", "Lint TypeScript code", () =>{
             let src = null;
+            const noNgFactory = filter([ "**/*.ts", "!**/*.ngfactory.ts" ]);
```
Better to add the ngfactory file type to the config constants. Also use the typescript glob from config as well e.g. config.globs.scripts.typescript, can't remember if this has **/ by default, but there is a setting for it I'm sure.

agxs committed:
Use constants for globs instead of hard-coded strings.

agxs committed:
Make sure the clean task removes ngfactories.

agxs committed:
Make sure the scripts-ngc task is synchronous to avoid race conditions.

This task must complete before the scripts-typescript task otherwise the
ngfactory files don't get compiled.
